lib/ansible/modules/source_control/gitlab_deploy_key.py

```diff
@@ -94,140 +107,216 @@
 
-    if not access_token and not private_token:
-        module.fail_json(msg="need either access_token or private_token")
+    if not HAS_GITLAB_PACKAGE:
+        module.fail_json(msg="Missing required gitlab module (check docs or install with: pip install python-gitlab")
 
-    input = {}
+    try:
+        gitlab_instance = gitlab.Gitlab(url=server_url, ssl_verify=verify_ssl, email=login_user, password=login_password,
+                                        private_token=login_token, api_version=4)
+        gitlab_instance.auth()
+    except (gitlab.exceptions.GitlabAuthenticationError, gitlab.exceptions.GitlabGetError) as e:
+        module.fail_json(msg="Failed to connect to Gitlab server: %s" % to_native(e))
+    except (gitlab.exceptions.GitlabHttpError) as e:
+        module.fail_json(msg="Failed to connect to Gitlab server: %s. \
+            Gitlab remove Session API now that private tokens are removed from user API endpoints since version 10.2." % to_native(e))
 
-    for key in ['title', 'key', 'can_push']:
-        input[key] = module.params[key]
+    gitlab_deploy_key = GitLabDeployKey(module, gitlab_instance)
 
-    success, existing = _find(module, api_url, project, input['key'], access_token, private_token)
+    project = findProject(gitlab_instance, project_identifier)
 
-    if not success:
-        module.fail_json(msg="failed to list deploy keys", result=existing)
+    if project is None:
+        module.fail_json(msg="Failed to create project: project %s doesn't exists" % project_identifier)
 
-    if existing:
-        input['id'] = existing['id']
+    deployKey_exists = gitlab_deploy_key.existsDeployKey(project, key_title)
 
-    changed = False
-    success = True
-    response = None
+    if state == 'absent':
+        if deployKey_exists:
+            gitlab_deploy_key.deleteDeployKey()
+            module.exit_json(changed=True, result="Successfully deleted deployKey %s" % key_title)
+        else:
+            module.exit_json(changed=False, result="DeployKey deleted or does not exists")
 
     if state == 'present':
-        if not existing or not _are_equivalent(existing, input):
-            if not module.check_mode:
-                success, response = _publish(module, api_url, project, input, access_token, private_token)
-            changed = True
-    else:
-        if existing:
-            if not module.check_mode:
-                success, response = _delete(module, api_url, project, existing['id'], access_token, private_token)
-            changed = True
+        if gitlab_deploy_key.createOrUpdateDeployKey(project, key_title, key_keyfile, {'can_push': key_can_push}):
 
-    if success:
-        module.exit_json(changed=changed, msg='Success', result=response, previous_version=existing)
-    else:
-        module.fail_json(msg='Failure', error=response)
+            module.exit_json(changed=True, result="Successfully created or updated the deployKey %s" % key_title,
```

Comment:
Please use `msg` instead of `result`. Also the standard casing for both parameters as well as return values is snail_case. So it would become `deploy_key` instead of `deployKey`.

Reply:
Will be fix in the next commit